### PR TITLE
treewide: apply codespell to the comments in source code

### PR DESCRIPTION
**Pull request by tchaikov**
for less spelling errors in comment.

### CODE REVIEW

File: utils/date.h [hunk 1]
```diff
@@ -25,7 +25,7 @@
 // SOFTWARE.
 //
 // Our apologies.  When the previous paragraph was written, lowercase had not yet
-// been invented (that woud involve another several millennia of evolution).
+// been invented (that would involve another several millennia of evolution).
```
Comment (mykaul) [inline utils/date.h:28]: This is not ours, so I opted to keep it as such.

Comment (tchaikov) [reply utils/date.h:28]: reverted.

**Comment by tchaikov** [comment]
v2:

- reverted the change in `utils/date.h`.

File: utils/date.h (hunk 1 above)
Comment (tchaikov) [reply utils/date.h]: will add this file to the skip list.

Comment (mykaul) [reply utils/date.h]: It was fixed years ago (https://github.com/HowardHinnant/date/commit/a811a20748d5354975a698ea6dc6e814146db08a ) which makes you wonder if we should not re-sync the code.

Comment (tchaikov) [reply utils/date.h]: guess the reason is that it is a low-priority working item to do. i will take a quick look to evaluate the risk and workload to update this file with the upstream.

**Comment by tchaikov** [comment]
> #### 🔴 CI State: FAILURE
> ✅ - [Build](https://jenkins.scylladb.com/job/scylla-master/job/scylla-ci/5234/) ✅ - [dtest](https://jenkins.scylladb.com/job/scylla-master/job/gating-dtest-release/6631/) ❌ - [Unit Tests](https://jenkins.scylladb.com/job/scylla-master/job/scylla-ci/5234/)
> 
> #### Failed Tests (3/23441):
> * [test_downgrade_after_partial_upgrade](https://jenkins.scylladb.com//job/scylla-master/job/scylla-ci/5234/testReport/junit/%28root%29/test_cluster_features/Tests___Unit_Tests___test_downgrade_after_partial_upgrade) [🔍](https://github.com/scylladb/scylladb/issues?q=is:issue+is:open+test_downgrade_after_partial_upgrade)
> * [topology.test_cluster_features.debug.1](https://jenkins.scylladb.com//job/scylla-master/job/scylla-ci/5234/testReport/junit/%28root%29/non-boost%20tests/Tests___Unit_Tests___topology_test_cluster_features_debug_1) [🔍](https://github.com/scylladb/scylladb/issues?q=is:issue+is:open+topology.test_cluster_features.debug.1)
> * [test_commitlog_reader](https://jenkins.scylladb.com//job/scylla-master/job/scylla-ci/5234/testReport/junit/boost/commitlog_test/test_commitlog_reader) [🔍](https://github.com/scylladb/scylladb/issues?q=is:issue+is:open+test_commitlog_reader)
> 
> #### Build Details:
> * Duration: 5 hr 29 min
> * Builder: monster

please check failing tests and file issues if they are new ones.

**Comment by tchaikov** [comment]
@mykaul Yaniv, could you help review this change? i see you just approved another one.

**Review by mykaul — APPROVED** [review]

**Review by denesb — APPROVED** [review]

**Comment by denesb** [comment]
I cannot queue because Avi queued https://github.com/scylladb/scylladb/pull/16450 (sorry, I was too slow to close it) and there will be conflicts. Please rebase once promotion runs.

**Comment by tchaikov** [comment]
no worries. lemme rebase it.

**Comment by tchaikov** [comment]
v3:

- rebased against master:HEAD

**Review by nyh — APPROVED** [review]

**Pull request closed without merging**